corehq/apps/app_manager/util.py

```diff
@@ -409,46 +409,83 @@ def _add_ancestors(ctype, generation):
     else:
         generations = []
 
+    subsets = [{
+        "id": generation_names[i],
+        "name": "{} ({})".format(generation_names[i], " or ".join(ctypes)) if i > 0 else base_case_type,
+        "key": "@case_type",
+        "structure": {p: {} for type in [map[t] for t in ctypes] for p in type},
+        "related": {"parent": {
+            "hashtag": "#case/" + generation_names[i + 1],
+            "subset": generation_names[i + 1],
+            "key": "@case_id",
+        }} if i < len(generations) - 1 else None,
+    } for i, ctypes in enumerate(generations)]
```

Comment:
This will show both parents if we can't determine which is which. It's confusing, but was the best we came up with with previous iteratoin of case management. I think it looks like `parent (mother, household)` in the form builder tree